WordPress/gutenberg
# Media & Text: Replace the deprecated __experimentalImageSizeControl with ResolutionTool

## What?

ImageSizeControl (__experimentalImageSizeControl) is used in the Media & text block. It was marked as deprecated in [#51545](https://github.com/WordPress/gutenberg/pull/51545/files#diff-5a9a2234d89e4080e031dbc85aafab7b248af84799f65710ad15ca7ca03bd5f1R34-R38).

This PR replaces __experimentalImageSizeControl with the ResolutionTool.

Partial for https://github.com/WordPress/gutenberg/issues/52223

## Why?

Deprecated controls should not be used.

## How?


Adds a new tools panel and moves the existing block sidebar options to this panel.
Replaces __experimentalImageSizeControl with the ResolutionTool.

## Testing Instructions





### Testing Instructions for Keyboard


Deactivate Gutenberg.
Add two new media and text blocks with images. One with default options, one with the "Crop image to fill entire column" option enabled and a non default resolution.
Save.

Activate Gutenberg with the PR applied. Refresh the page and confirm that:
-  There are no block validation errors.
- The options use a tools panel.
- All options, including the resolution, works.
- The reset and the "reset all" works.

Save and view the front of the website. Confirm that the blocks display correctly, and the correct image size is loaded.

## Screenshots or screencast 
Before:
<img width="284" alt="Media & text block with the settings panel opened, before the PR" src="https://github.com/WordPress/gutenberg/assets/7422055/60d68f88-8578-46ee-9a16-65a37862a9ad">

After:
<img width="284" alt="Media & text block with the settings panel opened, with the PR, using the tools item panels" src="https://github.com/WordPress/gutenberg/assets/7422055/2bd59705-3287-41b8-8fed-9703105fbeb9">

<img width="559" alt="Media & text block with the options panel opened." src="https://github.com/WordPress/gutenberg/assets/7422055/3e1b8abb-0c0b-4248-afd6-9520548f6ce5">
